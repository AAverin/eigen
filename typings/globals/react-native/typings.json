{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
