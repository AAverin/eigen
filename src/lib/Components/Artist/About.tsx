import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { Dimensions, StyleSheet, View } from "react-native"

import NavButton from "../Buttons/NavigationButton"
import RelatedArtists from "../RelatedArtists"
import Separator from "../Separator"
import Articles from "./Articles"
import Biography from "./Biography"

import { About_artist } from "__generated__/About_artist.graphql"

interface Props {
  artist: About_artist
}

class About extends React.Component<Props> {
  render() {
    return (
      <View>
        {this.biography()}
        {this.articles()}
        {this.relatedArtists()}
      </View>
    )
  }

  biography() {
    if (this.props.artist.has_metadata) {
      return (
        <View>
          <Biography artist={this.props.artist as any} />
          {this.auctionResults()}
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  auctionResults() {
    if (this.props.artist.is_display_auction_link) {
      // Keeps the same margins as the bio text
      const sideMargin = Dimensions.get("window").width > 700 ? 50 : 20
      const url = `/artist/${this.props.artist.slug}/auction-results`
      return (
        <NavButton title="Auction results" href={url} style={{ marginLeft: sideMargin, marginRight: sideMargin }} />
      )
    }
  }

  articles() {
    if (this.props.artist.articles.edges.length) {
      return (
        <View>
          <Articles articles={this.props.artist.articles.edges.map(({ node }) => node)} />
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  relatedArtists() {
    return this.props.artist.related.artists.edges.length ? (
      <RelatedArtists artists={this.props.artist.related.artists.edges.map(({ node }) => node)} />
    ) : null
  }
}

const styles = StyleSheet.create({
  sectionSeparator: {
    marginBottom: 20,
  },
})

export default createFragmentContainer(About, {
  artist: graphql`
    fragment About_artist on Artist {
      has_metadata: hasMetadata
      is_display_auction_link: isDisplayAuctionLink
      slug
      ...Biography_artist
      related {
        artists: artistsConnection(first: 16) {
          edges {
            node {
              ...RelatedArtists_artists
            }
          }
        }
      }
      articles: articlesConnection(first: 10) {
        edges {
          node {
            ...Articles_articles
          }
        }
      }
    }
  `,
})
