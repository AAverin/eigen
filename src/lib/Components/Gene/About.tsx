import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { StyleSheet, ViewStyle } from "react-native"

import Biography from "./Biography"

import RelatedArtists from "../RelatedArtists/RelatedArtists"
import Separator from "../Separator"

import { About_gene } from "__generated__/About_gene.graphql"
import { StickyTabPageScrollView } from "../StickyTabPage/StickyTabPageScrollView"

interface Props {
  gene: About_gene
}

const About: React.FC<Props> = ({ gene }) => {
  const relatedArtists = () => {
    return (gene.trending_artists || []).length ? <RelatedArtists artists={gene.trending_artists as any} /> : null
  }

  const biography = () => {
    return (
      <>
        <Biography gene={gene as any} style={styles.sectionSeparator} />
        <Separator style={styles.sectionSeparator} />
      </>
    )
  }

  return (
    <StickyTabPageScrollView contentContainerStyle={{ paddingTop: 15 }}>
      {biography()}
      {relatedArtists()}
    </StickyTabPageScrollView>
  )
}

interface Styles {
  sectionSeparator: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  sectionSeparator: {
    marginBottom: 20,
  },
})

export default createFragmentContainer(About, {
  gene: graphql`
    fragment About_gene on Gene {
      ...Biography_gene
      trending_artists: trendingArtists {
        ...RelatedArtists_artists
      }
    }
  `,
})
