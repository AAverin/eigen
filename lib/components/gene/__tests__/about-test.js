import 'react-native'
import React from 'react'
import About from '../about'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('../../opaque_image_view.js', () => 'ImageView')

it('renders correctly', () => {
  const gene = {
    description: 'This is some example text',
    trending_artists : []
  }
  const about = renderer.create(<About gene={gene}/>).toJSON()
  expect(about).toMatchSnapshot()
})

it('shows trending artists correctly', () => {
  const gene = {
    description: 'This is some example text',
    trending_artists : [{
      _id: '1',
      href: '/thing/artist/url',
      name: 'Artist Name',
      counts: {
        for_sale_artworks: 6,
        artworks: 6
      },
      image: {
        url: ''
      }
    }]
  }
  const about = renderer.create(<About gene={gene}/>).toJSON()
  expect(about).toMatchSnapshot()
})
