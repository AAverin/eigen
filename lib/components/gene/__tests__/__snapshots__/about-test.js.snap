exports[`test renders correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "marginLeft": 0,
          "marginRight": 0,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
              "marginBottom": 20,
            },
            Object {
              "fontFamily": "AGaramondPro-Regular",
            },
          ]
        }>
        This is some example text
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#e5e5e5",
            "height": 1,
            "marginLeft": -20,
            "marginRight": -20,
          },
          Object {
            "marginBottom": 20,
          },
        ]
      } />
  </View>
</View>
`;

exports[`test shows trending artists correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "marginLeft": 0,
          "marginRight": 0,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
              "marginBottom": 20,
            },
            Object {
              "fontFamily": "AGaramondPro-Regular",
            },
          ]
        }>
        This is some example text
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#e5e5e5",
            "height": 1,
            "marginLeft": -20,
            "marginRight": -20,
          },
          Object {
            "marginBottom": 20,
          },
        ]
      } />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flexDirection": "column",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "fontFamily": "AGaramondPro-Regular",
          },
        ]
      }>
      Related Artists
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-around",
          "marginLeft": -10,
          "marginRight": -10,
          "marginTop": 12,
        }
      }>
      <TouchableWithoutFeedback
        onPress={[Function]}>
        <View
          style={
            Object {
              "margin": 5,
              "paddingBottom": 20,
              "width": 157,
            }
          }>
          <ImageView
            imageURL=""
            style={
              Object {
                "height": 105,
                "width": 157,
              }
            } />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Avant Garde Gothic ITCW01Dm",
                "fontSize": 12,
                "marginTop": 10,
                "textAlign": "left",
              }
            }>
            ARTIST NAME
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666666",
                "fontFamily": "AGaramondPro-Regular",
                "fontSize": 16,
                "marginTop": 5,
              }
            }>
            6 works for sale
          </Text>
        </View>
      </TouchableWithoutFeedback>
      <View
        style={
          Object {
            "height": 105,
            "width": 157,
          }
        } />
    </View>
  </View>
</View>
`;
